### PR TITLE
first version of C# apim spec on autorest

### PR DESCRIPTION
**Pull request by solankisamir**
## Description
This is the first version of the API Management C# client library based on AutoRest spec. 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/ApiManagement/ApiManagement.Tests/Helpers/Extensions.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Xml.Linq;
+
+namespace ApiManagementManagement.Tests.Helpers
+{
+    public static class Extensions
+    {
+        public static bool DictionaryEqual<TKey, TValue>(this IDictionary<TKey, TValue> first, IDictionary<TKey, TValue> second)
```
Comment (shahabhijeet) [inline src/SDKs/ApiManagement/ApiManagement.Tests/Helpers/Extensions.cs]: @solankisamir 
nit: Can you not use Linq extension methods first.Except<String>(second) or better use Linq to do the comparison.
I assume you only want to compare keys and not their values, in that case you can still do that using the above

File: src/SDKs/ApiManagement/Management.ApiManagement/Microsoft.Azure.Management.ApiManagement.csproj [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <AssemblyTitle>Microsoft Azure API Management Management</AssemblyTitle>
+    <AssemblyName>Microsoft.Azure.Management.ApiManagement</AssemblyName>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <PackageTags>Microsoft Azure ApiManagement management;API Managementt;REST HTTP client;azureofficial;windowsazureofficial;netcore451511</PackageTags>
+  </PropertyGroup>
+  <PropertyGroup>
+    <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+  </PropertyGroup>
+  <ItemGroup>
+    <Folder Include="Customization\Models\" />
```
Comment (shahabhijeet) [inline src/SDKs/ApiManagement/Management.ApiManagement/Microsoft.Azure.Management.ApiManagement.csproj]: @solankisamir you don't need to include any folders. Project system will include it. This is not needed

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@solankisamir can you provide link to the Swagger PR that was merged.

File: src/SDKs/ApiManagement/ApiManagement.Tests/ApiManagement.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,59 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>ApiManagement.Tests</PackageId>
+    <Description>ApiManagement.Tests Class Library</Description>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
```
Comment (solankisamir) [inline src/SDKs/ApiManagement/ApiManagement.Tests/ApiManagement.Tests.csproj]: #fixed

File: src/SDKs/ApiManagement/ApiManagement.Tests/ApiManagement.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+  <!--Do not remove until VS Test Tools fixes #472-->
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+  <ItemGroup>
+    <Folder Include="Resources\" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Update="Resources\SwaggerPetStoreV2.json">
```
Comment (solankisamir) [inline src/SDKs/ApiManagement/ApiManagement.Tests/ApiManagement.Tests.csproj]: #fixed

File: src/SDKs/ApiManagement/ApiManagement.Tests/ApiManagementTestBase.cs [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+                    rgName = TestUtilities.GenerateName("sdktestrg");
+                    resourcesClient.ResourceGroups.CreateOrUpdate(rgName, new ResourceGroup { Location = this.location });
+                }
+                                
+                this.base64EncodedTestCertificateData = testEnv.ConnectionString.KeyValuePairs.GetValueUsingCaseInsensitiveKey(TestCertificateKey);
+                if (!string.IsNullOrEmpty(this.base64EncodedTestCertificateData))
+                {
+                    HttpMockServer.Variables[TestCertificateKey] = base64EncodedTestCertificateData;
+                }
+                this.testCertificatePassword = testEnv.ConnectionString.KeyValuePairs.GetValueUsingCaseInsensitiveKey(TestCertificatePasswordKey);
```
Comment (solankisamir) [inline src/SDKs/ApiManagement/ApiManagement.Tests/ApiManagementTestBase.cs]: #fixed

File: src/SDKs/ApiManagement/ApiManagement.Tests/ApiManagementTestBase.cs [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+                this.subscriptionId = testEnv.SubscriptionId;
+                HttpMockServer.Variables[TenantIdKey] = tenantId;
+                HttpMockServer.Variables[SubIdKey] = subscriptionId;                
+                HttpMockServer.Variables[ServiceNameKey] = this.serviceName;
+                HttpMockServer.Variables[LocationKey] = this.location;
+                HttpMockServer.Variables[ResourceGroupNameKey] = this.rgName;
+            }
+            else if (HttpMockServer.Mode == HttpRecorderMode.Playback)
+            {
+                this.tenantId = testEnv.Tenant;
```
Comment (solankisamir) [inline src/SDKs/ApiManagement/ApiManagement.Tests/ApiManagementTestBase.cs]: remove some extra variables. I require some initialization during playback, as some of the tests can only be executed on an active service. Service Activation takes 40 minutes, hence use this as a place to provide active service details, and proceed with additional validation.

File: src/SDKs/ApiManagement/Management.ApiManagement/Microsoft.Azure.Management.ApiManagement.csproj (hunk 2 above)
Comment (solankisamir) [inline src/SDKs/ApiManagement/Management.ApiManagement/Microsoft.Azure.Management.ApiManagement.csproj]: #fixed

**Comment by shahabhijeet** [comment]
@solankisamir as long as Kedar or someone from your team approves, this is good to merge

**Comment by solankisamir** [comment]
@shahabhijeet can you take a look?

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by KedarJoshi** [comment]
#sign-off

**Pull request merged**